runtime/src/commitment.rs

```diff
@@ -103,6 +103,22 @@ impl BlockCommitmentCache {
         self.highest_confirmed_slot
     }
 
+    pub fn highest_gossip_confirmed_slot(&self) -> Slot {
+        // TODO: see solana_core::RpcSubscriptions:
+        //self.last_checked_slots.get(&CommitmentLevel::SingleGossip).unwrap_or(&0)
+        self.highest_confirmed_slot
+    }
+
+    pub fn slot_with_commitment(&self, commitment_level: CommitmentLevel) -> Slot {
+        match commitment_level {
+            CommitmentLevel::Recent => self.slot(),
```

Comment:
Here's one change. This was `bank_forks.highest_slot()`

Reply:
This breaks `rpc::tests::test_get_vote_accounts`. I'll dig into why.

Reply:
Fixed. This PR makes working with RPC a little harder. Can't poke with BankForks without also poking at BlockCommitmentCache.